src/equation_of_state/MOM_EOS_linear.F90

```diff
@@ -512,13 +560,23 @@ subroutine int_spec_vol_dp_linear(T, S, p_t, p_b, alpha_ref, HI, Rho_T0_S0, &
     massWeight_bug = BTEST(MassWghtInterp, 3) ! True if the 8 bit is set
   endif
 
+  lambda = 0.0 ; if (dRho_dp/=0.0) lambda = 1.0 / dRho_dp
   do j=jsh,jeh ; do i=ish,ieh
     dp = p_b(i,j) - p_t(i,j)
-    dRho_TS = dRho_dT*T(i,j) + dRho_dS*S(i,j)
-    ! alpha_anom = 1.0/(Rho_T0_S0  + dRho_TS)) - alpha_ref
-    alpha_anom = ((1.0-Rho_T0_S0*alpha_ref) - dRho_TS*alpha_ref) / (Rho_T0_S0 + dRho_TS)
-    dza(i,j) = alpha_anom*dp
-    if (present(intp_dza)) intp_dza(i,j) = 0.5*alpha_anom*dp**2
+    p_ave = 0.5 * (p_t(i,j) + p_b(i,j))
+
+    drho = (dRho_dT * T(i,j) + dRho_dS * S(i,j)) + dRho_dp * p_ave
+    alpha_p_ave = 1.0 / (Rho_T0_S0 + drho)
+
+    eps = 0.5 * (dRho_dp * dp) * alpha_p_ave ; eps2 = eps * eps
+    ! alpha_anom = 1.0/(Rho_T0_S0 + dRho) - alpha_ref
+    alpha_anom = ((1.0 - Rho_T0_S0 * alpha_ref) - drho * alpha_ref) / (Rho_T0_S0 + drho)
+    ! The following expression would be more efficient but I suspect it changes answer.
+    ! alpha_anom = ((1.0 - Rho_T0_S0 * alpha_ref) - drho * alpha_ref) * alpha_p_ave
+    rem = (lambda * eps2) * (C1_3 + eps2 * (0.2 + eps2 * (C1_7 + C1_9 * eps2)))
```

Comment:
Is the expression for `rem` exact or is it a Taylor series expansion for the integral of the inverse of a polynomial in pressure.  If (as I suspect) it is the latter, please add a comment just before the expression for `rem` justifying the choice of the number of terms here, perhaps based on convergence to roundoff and a physically plausible range of `eps` for oceanic conditions.

Reply:
Done